apps/proxy/src/app/create-project/create-project.component.ts

```diff
@@ -0,0 +1,208 @@
+                    Validators.maxLength(20),
+                ],
+            ],
+            rateLimitHit: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            rateLimitMinute: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            selectedEnvironments: [''],
+        });
+
+        (this.destinationUrlForm = this.fb.group({
+            endpoint: ['', Validators.required],
+            ForwardUrl: this.fb.array([]),
+        })),
+            (this.gatewayUrlDetailsForm = this.fb.group({
+                useSameUrlForAll: [false],
+                gatewayUrls: this.fb.array([]),
+            }));
+    }
+
+    ngOnInit(): void {
+        this.getEnvironment();
+        this.projects$.pipe(takeUntil(this.destroy$)).subscribe((res) => {
+            if (res) {
+                const latestProject = res.data.slice().reverse();
+
+                latestProject.forEach((project) => {
+                    const baseUrl = `${environment.baseUrl}/proxy`;
+                    this.projectId = project.id;
+                    this.environments_with_slug = project.environments_with_slug.map((res) => ({
+                        name: res.name,
+                        url: `${baseUrl}/${this.projectId}/${res.project_slug}`,
+                    }));
+                });
+                this.populateGatewayUrls();
+                this.populateForwardUrls();
+            }
+        });
+        this.getProject$.pipe(takeUntil(this.destroy$)).subscribe((res) => {
+            if (res) {
+                this.currentstep = 2;
+                this.store.dispatch(rootActions.getAllProject());
+            }
+        });
+    }
+
+    get gatewayUrls(): FormArray {
+        return this.gatewayUrlDetailsForm.get('gatewayUrls') as FormArray;
+    }
+    get forwardUrls(): FormArray {
+        return this.destinationUrlForm.get('ForwardUrl') as FormArray;
+    }
+    populateGatewayUrls(): void {
+        this.gatewayUrls.clear();
+        this.environments_with_slug.forEach((env) => {
+            this.gatewayUrls.push(this.fb.control('', Validators.pattern(URL_REGEX)));
+        });
+    }
+    populateForwardUrls(): void {
+        this.forwardUrls.clear();
+        this.environments_with_slug.forEach((env) => {
+            this.forwardUrls.push(this.fb.control('', [Validators.required, Validators.pattern(URL_REGEX)]));
+        });
+    }
+    onChange(event: any): void {
+        if (event.checked) {
+            this.gatewayUrls.clear();
+            this.gatewayUrlDetailsForm.get('useSameUrlForAll').setValue(true);
+            this.gatewayUrlDetailsForm.addControl('singleUrl', this.fb.control(''));
+        } else {
+            this.gatewayUrlDetailsForm.removeControl('singleUrl');
+            this.gatewayUrlDetailsForm.get('useSameUrlForAll').setValue(false);
+            this.populateGatewayUrls();
+        }
+    }
+
+    checkInputfeild() {
```

Comment:
create a common function to change Step 
If something has to be done for a step do it in that function by checking currentStep
no need of two function here checkInputfeild and backStep
also check for spelling mistake in checkInputfeild